### PR TITLE
Search

### PR DESCRIPTION
**Pull request by wshangguan**
Added Book schema, API doc in swagger, and google search feature in Get(/api/books:id)

### CODE REVIEW

File: server/book/book.schema.ts [hunk 1]
```diff
@@ -11,23 +11,52 @@ export enum BookType {
...
   @Prop()
-  // google books' book id
-  gid: string;
+  gid: string; // google books' book id
 
   @Prop()
   isbn: string;
 
-  @Prop({
-    enum: [BookType.BOOK, BookType.MAGAZINE],
```
Comment (casrom) [inline server/book/book.schema.ts:21]: might still want to keep this enum?

File: package.json [hunk 2]
```diff
@@ -47,6 +47,8 @@
     "@types/passport": "^1.0.7",
     "@types/passport-auth0": "^1.0.5",
     "axios": "^0.24.0",
+    "class-transformer": "^0.4.0",
```
Comment (hans-lizihan) [inline package.json:50]: what are those for?

File: server/book/book.controller.ts [hunk 3]
```diff
@@ -6,44 +6,62 @@ import {
...
-  // return a book in our database that matchs a gid
   @Get(':id')
+  @ApiTags('books')
+  @ApiOperation({
+    summary: "Find a book's detail by gid",
+    description:
+      'id refers to Google Books gid, check cache in db first, query Google Books API if not found in db.',
+  })
+  @ApiResponse({
+    status: 500,
```
Comment (hans-lizihan) [inline server/book/book.controller.ts:48]: i think it's more appropriate to give a 404 here

File: tsconfig.build.tsbuildinfo [hunk 4]
```diff
@@ -0,0 +1 @@
...egory":1,"code":2307}]],545,536,[537,[{"file":"./server/auth/auth.module.ts","start":119,"length":18,"messageText":"Cannot find module 'user/user.module' or its corresponding type declarations.","category":1,"code":2307}]],532,604,[534,[{"file":"./server/auth/auth0.strategy.ts","start":191,"length":19,"messageText":"Cannot find module 'user/user.service' or its corresponding type declarations.","category":1,"code":2307}]],605,[535,[{"file":"./server/auth/passport.serializer.ts","start":158,"length":18,"messageText":"Cannot find module 'user/user.schema' or its corresponding type declarations.","category":1,"code":2307},{"file":"./server/auth/passport.serializer.ts","start":206,"length":19,"messageText":"Cannot find module 'user/user.service' or its corresponding type declarations.","category":1,"code":2307},{"file":"./server/auth/passport.serializer.ts","start":626,"length":4,"messageText":"Parameter 'user' implicitly has an 'any' type.","category":1,"code":7006},{"file":"./server/auth/passport.serializer.ts","start":667,"length":5,"messageText":"Parameter 'error' implicitly has an 'any' type.","category":1,"code":7006}]],543,541,542,544,546,603,606,540,538,539]},"version":"4.4.4"}
```
Comment (hans-lizihan) [inline tsconfig.build.tsbuildinfo:1]: please remove this file. i think i had it configured in `.gitignore`, did you merge with main?

**Review by hans-lizihan — APPROVED** [review]

**Pull request merged**